Xplat/src/main/java/vazkii/botania/data/LooniumStructureLootProvider.java

```diff
@@ -0,0 +1,255 @@
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.TRAIL_RUINS_ARCHAEOLOGY_RARE))
+				)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_PLAINS),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_PLAINS_HOUSE, PLAINS_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_DESERT),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_DESERT_HOUSE, DESERT_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_SAVANNA),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_SAVANNA_HOUSE, SAVANNA_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_SNOWY),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_SNOWY_HOUSE, SNOWY_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_TAIGA),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_TAIGA_HOUSE, TAIGA_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.WOODLAND_MANSION),
+				LootTable.lootTable().withPool(LootPool.lootPool()
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.WOODLAND_MANSION).setWeight(99))
+						.add(LootItem.lootTableItem(Items.TOTEM_OF_UNDYING).setWeight(1))
+				)
+		);
+
+		var output = new ArrayList<CompletableFuture<?>>(tables.size());
+		for (var e : tables.entrySet()) {
+			Path path = pathProvider.json(e.getKey());
+			LootTable.Builder builder = e.getValue();
+			LootTable lootTable = builder.setParamSet(LootContextParamSets.EMPTY).build();
+			JsonElement jsonTree = Deserializers.createLootTableSerializer().create().toJsonTree(lootTable);
+			output.add(DataProvider.saveStable(cache, jsonTree, path));
+		}
+		return CompletableFuture.allOf(output.toArray(CompletableFuture[]::new));
+	}
+
+	private static LootTable.Builder buildVillageLootTable(ResourceLocation house, Set<VillageLoot> villageLootSet) {
+		LootPool.Builder lootPool = LootPool.lootPool().add(LootTableReference.lootTableReference(house).setWeight(3));
+		for (var loot : villageLootSet) {
+			lootPool.add(LootTableReference.lootTableReference(loot.lootTable));
+		}
+		return LootTable.lootTable().withPool(lootPool);
+	}
+
+	@NotNull
+	private static LootTable.Builder buildShipwreckLootTable() {
+		return LootTable.lootTable().withPool(LootPool.lootPool()
+				.add(LootTableReference.lootTableReference(BuiltInLootTables.SHIPWRECK_MAP))
+				.add(LootTableReference.lootTableReference(BuiltInLootTables.SHIPWRECK_SUPPLY))
+				.add(LootTableReference.lootTableReference(BuiltInLootTables.SHIPWRECK_TREASURE))
+		);
+	}
+
+	@NotNull
+	private static LootTable.Builder buildDelegateLootTable(ResourceLocation reference) {
+		return LootTable.lootTable().withPool(LootPool.lootPool()
+				.add(LootTableReference.lootTableReference(reference))
+		);
+	}
+
+	@NotNull
+	private static LootTable.Builder buildOceanRuinLootTable(ResourceLocation archaeology) {
+		// Note: since the Loonium does not supply a location, treasure maps will roll as empty maps
```

Comment:
Could we make it supply a location? would be cool

Reply:
Using the mob's death location sounds fun, then you could bring them with you places to roll treasure maps for that area

Reply:
Then let's shelve that idea for now and maybe come back to it later